### ISSUE TITLE
.fill for author with sections=['publications'] seems to have stopped working on 7/8/2021

### ISSUE THREAD

**Issue by BISCoder** [scholarly-python-package/scholarly#297]
My program was processing fine until morning 7/8/2021. 

Then Scholarly started giving me the error:  <class 'KeyError'> on calls such as resp = scholarly.fill(q),  where q is an author object newly returned from a search_author_id(q) where q was a GoogleScholarID as shown (zzz) in a web page https://scholar.google.com/citations?user=zzz&hl=en
This had been working wonderfully before.

I tried calling fill(author, sections=['basics', 'indices', 'coauthors', 'counts']  that was fine but [] and 'publications' both failed. 

I am hoping to have this work again soon. In any case thanks for your attention.

- b

**Comment by ipeirotis** [comment]
Please give a snippet of the code that causes an error. 

I would recommend trying it on Colab using a proxy, and see if the issue persists. If not, then it is likely an issue of being blocked by Google.

**Comment by BISCoder** [comment]
Many thanks for the prompt reply.
Here is a snippet that fails. 
This is a "new" google id I had never tried before.
I tried it from one IP address, and then I compiled it and ran the same thing on a computer using a vpn.
I have not tried Colab and have been metering my queries rather than using a proxy so far. I can look into that.
Thanks again for your good code and for looking at this.

from scholarly import scholarly
from scholarly.data_types import Publication # I know, SQL is a crutch... :)
import sys
gid = 'Iin2LSIAAAAJ'
Auth = scholarly.search_author_id(gid)
try:
    resp = scholarly.fill(Auth)
except: # catch *all* exceptions
    e = sys.exc_info()[0]
    print(f' error doing query{e}')

print('tested scholarly call')

**Comment by BISCoder** [comment]
It looks like the interface removed the tabs from the code. Is that ok?  :)

**Comment by joan-navarro** [comment]
Hi!

I have seen that replacing the lines 110 and 111 of the file `publication_parser.py`: 

```
publication['author_pub_id'] = re.findall(_CITATIONPUBRE, __data.find(
'a', class_='gsc_a_at')['data-href'])[0]
```

with:

`publication['author_pub_id']=re.findall(_CITATIONPUBRE, str(__data))[0]`

shall fix the problem.

Hope it helps!

**Comment by BISCoder** [comment]
Thanks again for the quick response. I guess that means I would need to interact with the scholarly source code? That will be a new adventure for me.

**Comment by joan-navarro** [comment]
> 
> 
> Thanks again for the quick response. I guess that means I would need to interact with the scholarly source code? That will be a new adventure for me.

It's not a big deal, you only need to find where the file "publication_parser.py" is in your system and update it as I wrote above :)

**Comment by BISCoder** [comment]
Thanks, Cool! it was easy to do. Your advice helped.
With that change, I no longer get that error.
However, when I then called fill for the pub I get a different error "<class 'AttributeError'>"
Here is my updated snippet:
from scholarly import scholarly
from scholarly.data_types import Publication # I know, SQL is a crutch... :)
import sys

gid = 'Iin2LSIAAAAJ'
Auth = scholarly.search_author_id(gid)
resp = scholarly.fill(Auth)
for pub in resp['publications']:
    try:
        resp = scholarly.fill(pub)
        print(resp)
    except: # catch *all* exceptions
        e = sys.exc_info()[0]
        print(f' error doing query{e}')

print('tested scholarly call')
Again - I really appreciate your help, and, being new to this environment, pleas let me know if I am out of line in a request.
- b

**Comment by arcadianlyric** [comment]
> Hi!
> 
> I have seen that replacing the lines 110 and 111 of the file `publication_parser.py`:
> 
> ```
> publication['author_pub_id'] = re.findall(_CITATIONPUBRE, __data.find(
> 'a', class_='gsc_a_at')['data-href'])[0]
> ```
> 
> with:
> 
> `publication['author_pub_id']=re.findall(_CITATIONPUBRE, str(__data))[0]`
> 
> shall fix the problem.
> 
> Hope it helps!

Hi,

I replaced the code, but still got the same 'data-href' error. 
Your functions are very helpful to my analysis, I'm hoping to resolve the issue soon.
Thank you.

**Comment by BISCoder** [comment]
I want to say again that the previous fix worked for me (thanks joan-navarro), here is a bit more info on the second problem:

I would like to note that I believe the problem is as shown in the middle line here from the fill function in publication_parser(approximately line 267)
Snippet:
            print('before possible error site', end = '')
            publication['bib']['title'] = soup.find('div', id='gsc_vcd_title').text    #This line has the problem
            print('after possible error site')
When I run the snippet with these print command in place, I get: 
before possible error site error doing query<class 'AttributeError'>
'after possible error site' is not printed.
I think that may mean that there is not a 'gsc_vcd_title' division in the page any more. Google may have altered the representation?
Not an expert, just thinking on the issue.

**Comment by BISCoder** [comment]
Doing a little more, I discovered that changing  '_vcd_' to '_oci_' in provided constants throughout the fill function in the publication_parser.py file made it work, at lest in my application. I expect there are more elegant solutions but it got my program running.
Many thanks for the suggestions - without the help I could not have made any progress.
Meanwhile, perhaps others need this fix? I don't know enough to say. :)

- b

**Comment by Aniruth-ai** [comment]
Hello guys,

I am also facing the same issue as mentioned in this chain. For that to resolve, I made the changes mentioned by @arcadianlyric. However I am getting these errors after making the changes:

**urllib.error.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1108)>**

**AttributeError: 'ProxyGenerator' object has no attribute '_webdriver'**

Any idea to resolve this issue will be appreciated.

Thanks!

**Comment by ipeirotis** [comment]
Issue fixed in v1.2.1a
